### PR TITLE
[Cisco ASA] message 716059 alternative format

### PR DESCRIPTION
**Pull request by jrmolin**
## Proposed commit message

[cisco asa] Message 716059 has new format

- Add new grok patterns to capture the new message structure
- Verify no regressions

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## How to test this PR locally

```
cd packages/cisco_asa
elastic-package test
```

## Related issues

-

### CODE REVIEW

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -944,6 +944,7 @@ processors:
       patterns:
         - '^Group <%{NOTBRACKET:_temp_.cisco.webvpn.group_name}> User <%{NOTBRACKET:source.user.name}> IP <%{NOTBRACKET:destination.address}> AnyConnect session resumed. Connection from %{NOTSPACE:source.address}\.$'
         - '^Group %{NOTSPACE:_temp_.cisco.webvpn.group_name} User %{NOTSPACE:source.user.name} IP %{NOTSPACE:destination.address} AnyConnect session resumed. Connection from %{NOTSPACE:source.address}\.$'
+        - '^Group <%{NOTBRACKET:_temp_.cisco.webvpn.group_name}> User( <%{NOTBRACKET:source.user.name}>)? IP <%{NOTBRACKET:destination.address}> AnyConnect session resumed connection from IP <%{NOTBRACKET:source.address}>\.$'
```
Comment (pkoutsovasilis) [inline packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:947]: this one can be merged with the first one in the following fashion

`'^Group <%{NOTBRACKET:_temp_.cisco.webvpn.group_name}> User( <%{NOTBRACKET:source.user.name}>)? IP <%{NOTBRACKET:destination.address}> AnyConnect session (resumed connection|resumed. Connection) from IP <%{NOTBRACKET:source.address}>\.$'`

 right?! maybe I miss something

Comment (taylor-swanson) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:947]: I didn't even notice `AnyConnect session resumed` versus `AnyConnect session resumed connection from IP`.

In the `<>` case, the last IP address should be wrapped in `<>` as well, so Panos' suggestion should be correct. (The omission of `<>` was my mistake when I added it).

**Comment by taylor-swanson** [comment]
Title nit: I don't think we should mention SDH in the public integrations repo, it's not going to mean anything to anyone else.

Something like this is fine:

`[cisco_asa] Message 716059 alternative format`

I generally follow the format of:

`[integration] Short description of change`

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (pkoutsovasilis) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: @taylor-swanson based on this 
>In the <> case, the last IP address should be wrapped in <> as well, so Panos' suggestion should be correct. (The omission of <> was my mistake when I added it).

does this mean that the following log line in our test logs requires a wrapped in `<>` IP at the end
><140>Oct 03 2023 16:40:40 myAsaHostname : %ASA-6-716059: Group <GROUP 1> User <USER 1> IP <10.20.0.1> AnyConnect session resumed. Connection from 172.16.0.1.

Comment (taylor-swanson) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: Yes, we'll need to update that line (that wasn't a "real" log, but rather one I created based on the patterns I was seeing elsewhere. I just forgot about the IP at the end).

**Review by taylor-swanson — APPROVED** [review]

**Pull request merged**